cmd/package-server-manager/main.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	configv1 "github.com/openshift/api/config/v1"
+	olmv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	controllers "github.com/openshift/operator-framework-olm/pkg/package-server-manager"
+	//+kubebuilder:scaffold:imports
+)
+
+const (
+	defaultName                 = "packageserver"
+	defaultNamespace            = "openshift-operator-lifecycle-manager"
+	leaderElectionConfigmapName = "packageserver-controller-lock"
+)
+
+func setupScheme() *runtime.Scheme {
+	scheme := runtime.NewScheme()
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(configv1.Install(scheme))
+	utilruntime.Must(olmv1alpha1.AddToScheme(scheme))
+
+	return scheme
+}
+
+func main() {
+	cmd := newStartCmd()
+
+	if err := cmd.Execute(); err != nil {
+		fmt.Fprintf(os.Stderr, "encountered an error while executing the binary: %v", err)
+		os.Exit(1)
+	}
+}
+
+func run(cmd *cobra.Command, args []string) error {
+	name, err := cmd.Flags().GetString("name")
+	if err != nil {
+		return err
+	}
+	namespace, err := cmd.Flags().GetString("namespace")
+	if err != nil {
+		return err
+	}
+	disableLeaderElection, err := cmd.Flags().GetBool("disable-leader-election")
+	if err != nil {
+		return err
+	}
+
+	opts := zap.Options{
+		Development: true,
+	}
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	setupLog := ctrl.Log.WithName("setup")
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), manager.Options{
+		Scheme:                  setupScheme(),
+		Namespace:               namespace,
+		MetricsBindAddress:      "0",
```

Comment:
This is likely missing a TODO comment about re-enabling metrics unless there's any reason we wouldn't want to expose that functionality in the container.

Reply:
Ah, interesting it looks like we also disable this currently for OLM as well looking at the staging/operator-lifecycle-manager/cmd/olm main package(s).

Reply:
I ended up enabling this and moved the default port value to 8083 to avoid conflicting with the OLM Operator metric ports (8080-8082). Likely another reason why it would be easier to deploy this controller as a standalone deployment vs. a sidecar container, but we can revisit this approach later in the 4.9 development cycle.